lib/matplotlib/tests/test_scale.py

```diff
@@ -210,3 +212,9 @@ def test_pass_scale():
     ax.set_yscale(scale)
     assert ax.xaxis.get_scale() == 'log'
     assert ax.yaxis.get_scale() == 'log'
+
+
+def test_scale_deepcopy():
+    sc = mscale.LogScale(axis='x', base=10)
+    sc2 = copy.deepcopy(sc)
+    assert str(sc.get_transform()) == str(sc2.get_transform())
```

Comment:
As a rough check that there is no shared state, I'd add
~~~
assert sc._transform is not sc2._transform
~~~
or, if you're uncomfortable with testing private attributes:
~~~
assert sc.get_transform() is not sc2.get_transform()
~~~
But personally, I'd test the internal state here, because `get_transform()` *could* be written to return copies, so that testing `get_transform()` include an additional assumption.

Reply:
yes

Reply:
:+1: The test found an implementation flaw. It's working :smile: